github_api_toolkit/__init__.py

```diff
@@ -155,3 +153,162 @@ def post(self, url: str, params: dict = {}, add_prefix: bool = True) -> requests
+                            text
+                        }
+                    }
+                }
+            }
+        '''
+
+        self.params = {
+            'org': org,
+            'repo': repo
+        }
+
+        response = self.make_ql_request(self.query, self.params)
+
+        if response.status_code == 200:
+            codeowners_contents = response.json()["data"]["repository"]["file"]["text"]
+
+            codeowners = codeowners_contents.split("\n")
+
+            # Remove any empty strings from the list of codeowners
+
+            indexes_removed = 0
+
+            for i in range(0, len(codeowners)):
+                if codeowners[i - indexes_removed] == "":
+                    codeowners.pop(i - indexes_removed)
+                    indexes_removed += 1
+
+            for i in range(len(codeowners)):
+                # If the codeowner contains a "/", it is a team and needs to be split from the organisation name.
+                # If there is no "/", it is a user and needs to have the @ symbol removed.
+                if "/" in codeowners[i]:
+                    codeowners[i] = codeowners[i].split("/")[-1]
+                else:
+                    codeowners[i] = codeowners[i].replace("@", "")
+
+            return codeowners
+        else:
+            response_json = response.json()
+            return response_json["message"], response_json["status"]
+        
+    def get_team_maintainers(self, org: str, team_name: str) -> list | tuple:
+        """Gets the maintainers of a GitHub team.
+
+        Args:
+            org (str): the GitHub organization name.
+            team_name (str): the GitHub team name.
+
+        Returns:
+            list | tuple: A list of maintainers in the team or a tuple containing an error message and status code.
+        """
+
+        self.query = '''
+            query ($org: String!, $team_name: String!) {
+                organization(login: $org) {
+                    team(slug: $team_name) {
+                        members(role: MAINTAINER) {
+                            nodes {
+                                login
+                            }
+                        }
+                    }
+                }
+            }
+        '''
+
+        self.params = {
+            'org': org,
+            'team_name': team_name
+        }
+
+        response = self.make_ql_request(self.query, self.params)
+
+        if response.status_code == 200:
+            try:
+                return response.json()["data"]["organization"]["team"]["members"]["nodes"]
+            except TypeError:
+                # If the above raises a TypeError, it means that the team_name is not a team, but a user.
+                # Therefore we should return the team_name as a list.
+                return [{"login": team_name}]
+        else:
+            response_json = response.json()
+            return response_json["message"], response_json["status"]
```

Comment:
Comment as above.

Since you have multiple cases where you check the same thing it might be nice to have a helper function so that you aren't repeating the same code.

def get_error_message(response_json):
    return response_json.get("message", "Unknown error"), response_json.get("status", "Unknown status")

return get_error_message(response.json())